src/main/java/sirius/db/es/constraints/ElasticFilterFactory.java

```diff
@@ -397,4 +401,44 @@ public ElasticConstraint constantScore(ElasticConstraint constraint, float boost
 
         return new ElasticConstraint(jsonObject);
     }
+
+    /**
+     * Creates a "dis_max" query.
+     * <p>
+     * This is mostly equivalent to {@link #or(List)}. However, instead of summing the scores of each child clause,
+     * the best (max) is picked.
+     *
+     * @param constraints the constraint cluases to add (at leas one has to match in order for the query to match)
+     * @return the newly created constraint or <tt>null</tt> if the list was empty or did only contain <tt>null</tt>
+     * constraints
+     */
+    @Nullable
+    public ElasticConstraint maxScore(@Nonnull List<ElasticConstraint> constraints) {
+        List<ElasticConstraint> effectiveConstraints = constraints.stream().filter(Objects::nonNull).toList();
+        if (effectiveConstraints.isEmpty()) {
+            return null;
+        }
+
+        if (effectiveConstraints.size() == 1) {
+            return constraints.get(0);
+        }
+
+        ArrayNode queries = Json.createArray();
+        effectiveConstraints.stream().filter(Objects::nonNull).forEach(q -> queries.add(q.toJSON()));
```

Comment:
```suggestion
        effectiveConstraints.forEach(q -> queries.add(q.toJSON()));
```

Reply:
Would be nice if `q` was renamed to `constraint`